### PR TITLE
[Tooling] Integrate Tilt development environment 

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Integrate [Tilt]() (and k8s/helm) development environment.

## Issue

- https://github.com/pokt-network/poktroll/issues/873

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: tiltfiles/k8s/gql-engine/templates/gql-engine-svc.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  name: gql-engine-service
+spec:
+  selector:
+    app.kubernetes.io/name: gql-engine
+    app.kubernetes.io/part-of: pocketdex
+    app.kubernetes.io/managed-by: helm
+  ports:
+    - name: http
+      protocol: TCP
+      port: 3000
```
Comment (bryanchriswhite) [inline tiltfiles/k8s/gql-engine/templates/gql-engine-svc.yaml:13]:
```suggestion
      port: {{ .Values.port }}
```

File: tiltfiles/k8s/gql-engine/templates/gql-engine-svc.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      targetPort: 3000
```
Comment (bryanchriswhite) [inline tiltfiles/k8s/gql-engine/templates/gql-engine-svc.yaml:14]:
```suggestion
      port: {{ .Values.port }}
      targetPort: {{ .Values.port }}
```

**Comment by Olshansk** [comment]
@bryanchriswhite This is `in progress` but also `ready for review` and `has no reviewers` and requires an investigation in [1].

Can you summarize the next step/blocker?

[1] https://github.com/tilt-dev/tilt-extensions/tree/master/git_resource

**Comment by bryanchriswhite** [comment]
@Olshansk the `TODO_INVESTIGATE` is not urgent and can be committed for now. I've requested review from @okdas, and @jorgecuesta and @Alann27.

**Review by okdas — APPROVED** [review]
`gco chore/tilt && tilt up` worked from the first try 😍 Thank you!

LGTM. I do think we need to document both `tilt up` and `tilt down` somewhere, though. How about `README.md`?

**Comment by bryanchriswhite** [comment]
@okdas yes, we should definitely document how to use tilt. My thinking was to do that in a subsequent PR, in which we also remove the docker compose based environments and scripts. The immediate intention here is to support https://github.com/pokt-network/poktroll/pull/885.

**Comment by bryanchriswhite** [comment]
@okdas I added a section to the README and made it "parallel" with the docker-compose option.
